### PR TITLE
Add documentation for postgres WITH clause

### PR DESCRIPTION
**Pull request by warrenseymour**
Documents the `.with()` query builder method introduced by #239

### CODE REVIEW

File: api.html [hunk 1]
```diff
@@ -876,6 +921,21 @@
...
+            flavours: ['postgres'],
+            desc: 'Combine with another <code>SELECT</code>, <code>UPDATE</code>, <code>INSERT</code> or <code>DELETE</code> using a Common Table Expression (CTE), ie a <code>WITH</code> clause',
+            params: {
+              alias: {
+                desc: 'The alias that the table expression should use',
+                type: 'string'
+              },
+              table: {
+                desc: 'Another <code>SELECT</code>, <code>UPDATE</code>, <code>INSERT</code> or <code>DELETE</code> query',
+                type: 'QueryBuilder'
```
Comment (hiddentao) [inline api.html]: Should be `'[[cls.QueryBuilder]]'`

File: api.html [hunk 2]
```diff
@@ -516,6 +531,21 @@
...
```
Comment (hiddentao) [inline api.html]: Should be `'[[cls.QueryBuilder]]'`

File: api.html [hunk 3]
```diff
@@ -345,6 +345,21 @@
...
```
Comment (hiddentao) [inline api.html]: Should be `'[[cls.QueryBuilder]]'`

File: api.html [hunk 4]
```diff
@@ -713,6 +743,21 @@
...
```
Comment (hiddentao) [inline api.html]: Should be `'[[cls.QueryBuilder]]'`

**Comment by hiddentao** [comment]
Thanks! Please make the following replacements:

* `type: 'string'` -> `type: 'String'`
* `type: 'QueryBuilder'` -> `type: '[[cls.QueryBuilder]]'`

For the descriptions which say `Combine with another Select, Delete, etc..` would it be easier if they simply said `Combine with another query...`...?

**Comment by warrenseymour** [comment]
@hiddentao Done

**Comment by hiddentao** [comment]
Thanks!

**Pull request merged**